pkg/csls/stream.go

```diff
@@ -81,3 +89,11 @@ func (w *Stream) PutCloudfoundryLog(log cloudfoundry.Log, groupName string) erro
 	}
 	return nil
 }
+
+func gzip_compress(in []byte) []byte {
+	var out bytes.Buffer
+	gz := gzip.NewWriter(&out)
+	gz.Write(in)
```

Comment:
return values need to be checked. using `Write` manually is a bit of a pain, I'd avoid it if possible (see other comment)